engine/src/main/java/com/datatorrent/stram/StreamingAppMasterService.java

```diff
@@ -649,7 +649,7 @@ protected void serviceStart() throws Exception
       addSSLConfigResource(config);
       WebApp webApp = WebApps.$for("stram", StramAppContext.class, appContext, "ws").with(config).start(new StramWebApp(this.dnmgr));
       LOG.info("Started web service at port: " + webApp.port());
-      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getHostName() + ":" + webApp.port();
+      appMasterTrackingUrl = NetUtils.getConnectAddress(webApp.getListenerAddress()).getAddress().getCanonicalHostName() + ":" + webApp.port();
```

Comment:
@vrozov in your example you get `vrozov.local` and not `vrozov.mydomain.com` (or whatever applies to your host) as required in the JIRA. What do you need to do on your machine to make it return vrozov.mydomain.com (using either `getHostName()` or `getCanonicalHostName()`)

I also added a comment in the JIRA regarding using wildcard or SANs in the certificate being the main issue rather than this